api/net/Areas/Subscriber/Controllers/ReportController.cs

```diff
@@ -155,34 +155,23 @@ public IActionResult GetPublicReports()
     /// If request includes 'generate=true' then make sure to return an unsent instance.
     /// </summary>
     /// <param name="id"></param>
-    /// <param name="generate">If an unsent instance does not exist, generate one.</param>
     /// <returns></returns>
     [HttpGet("{id}")]
     [Produces(MediaTypeNames.Application.Json)]
     [ProducesResponseType(typeof(ReportModel), (int)HttpStatusCode.OK)]
     [ProducesResponseType(typeof(ErrorResponseModel), (int)HttpStatusCode.BadRequest)]
     [SwaggerOperation(Tags = new[] { "Report" })]
-    public async Task<IActionResult> FindByIdAsync(int id, bool generate = false)
+    public IActionResult FindById(int id)
     {
         var username = User.GetUsername() ?? throw new NotAuthorizedException("Username is missing");
         var user = _userService.FindByUsername(username) ?? throw new NotAuthorizedException("User does not exist");
 
         var report = _reportService.FindById(id) ?? throw new NoContentException();
         if (report.OwnerId != user.Id) throw new NotAuthorizedException("Not authorized to view this report");
 
-        var instance = _reportService.GetLatestInstance(id, user.Id, false);
-        if (generate && instance == null)
-        {
-            // Generate a new instance of this report.
-            instance = await _reportHelper.GenerateReportInstanceAsync(new Services.Models.Report.ReportModel(report, _serializerOptions), user.Id);
-            _reportInstanceService.AddAndSave(instance);
-            instance = _reportService.GetLatestInstance(report.Id, user.Id, false) ?? throw new NoContentException("Report does not exist");
-        }
-        if (instance != null)
-        {
-            report.Instances.Clear();
-            report.Instances.Add(instance);
-        }
+        var instances = _reportService.GetLatestInstances(id, user.Id);
```

Comment:
Simplified this endpoint.  Use the `generate` endpoint to generate a new instance.